nilearn/decomposition/multi_pca.py

```diff
@@ -244,16 +257,42 @@ def fit(self, imgs=None, y=None, confounds=None):
 
             for param_name in ['target_affine', 'target_shape',
                                'smoothing_fwhm', 'low_pass', 'high_pass',
-                               't_r', 'memory', 'memory_level']:
+                               't_r', 'memory']:
                 our_param = getattr(self, param_name)
                 if our_param is None:
-                    # Default value
-                    continue
-                if getattr(self.masker_, param_name) is not None:
-                    warnings.warn('Parameter %s of the masker overriden'
-                                  % param_name)
+                    continue  # nothing to change
+                their_param = getattr(self.masker_, param_name)
+                if their_param is not None:
+                    # This block: conflict reporting & resolution
+                    if param_name != 'memory' and our_param == their_param:
+                        continue  # no conflict; values match.
+                    if param_name == 'memory':
+                        # Memory takes a more complex comparison and setting
+                        their_memory_level = getattr(self.masker_, 'memory_level')
+                        our_memory_level = getattr(self, 'memory_level')
+                        if is_same_memory_settings(our_param, their_param,
+                                                   our_memory_level,
+                                                   their_memory_level):
+                            continue  # no conflict, memory value matches
+                        elif their_memory_level != our_memory_level:
+                            # Must set memory_level when memory differs,
+                            #   as it is not independently in the list.
+                            setattr(self.masker_, 'memory_level', our_memory_level)
```

Comment:
I still don't understand why you are making a special case of `memory_level`. Here you silently change it which is, for me, not a behavior that we want. Could you elaborate on that?